<?xml version="1.0"?>

<launch>


  <!-- Starts gazebo with El Estero world, spawns Kingfisher, etc -->
  <arg name="world" default="elestero.world" />
  <include file="$(find kingfisher_gazebo)/launch/kingfisher_gazebo_worlds.launch">
    <arg name="world" value="$(arg world)"/>
    <!--<arg name="world" value="elestero_potentials.world"/>-->
  </include>

  <!-- Takes output from P3D Gazebo plugin to simulate Microstrain Nav. -->
  <node pkg="kingfisher_gazebo" type="p3d2nav.py" name="p3d2nav">
    <param name="gazebo_origin_lat" value="36.596524" type="double" />
    <param name="gazebo_origin_long" value="-121.888169" type="double" />
  </node>

  <!-- Converts Lat/Lon from Microstrain to local X/Y -->
  <node pkg="geonav_transform" type="geonav_transform_node" name="geonav_transform_node" clear_params="true" output="screen">
    <!-- Datum as latitude, longitude [decimal deg.], yaw [ENU, degrees] -->
    <rosparam param="datum">[36.595, -121.89, 0.0]</rosparam>
    <param name="base_link_frame_id" value="base_footprint" />
  </node>

</launch>
